.github/workflows/deploy-universal.yml

```diff
@@ -0,0 +1,102 @@
+name: Deployment (Universal)
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to deploy to'
+        required: true
+        default: 'stage'
+        type: choice
+        options: ['stage', 'prod']
+      delete-volumes:
+        description: 'Delete volumes?'
+        type: boolean
+        default: false
+      restore-db:
+        description: 'Restore DB?'
+        type: boolean
+        default: false
+      flush-redis:
+        description: 'Clear redis?'
+        type: boolean
+        default: false
+      build-flag:
+        type: choice
+        description: docker-compose additional build flag
+        default: "None"
+        options:
+          - "None"
+          - "--no-cache"
+
+jobs:
+  deployment:
+    concurrency: ${{ github.event.inputs.environment }}
+    runs-on: [self-hosted, "${{ github.event.inputs.environment }}"]
+    environment: ${{ github.event.inputs.environment == 'stage' && 'Staging' || 'Production' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Verify environment
+        run: |
+          (diff -wB <(grep -vE '^\s*#' /opt/deployment-specific-files/.env | cut -d '=' -f 1 | sort) \
+          <(grep -vE '^\s*#' .env.sample | cut -d '=' -f 1 | sort)) ||\
+          (echo "Environment variables mismatch" &&\
+          echo "Update /opt/deployment-specific-files/.env on the host or .env.sample in your branch" &&\
+          exit 1)
+
+      - name: Copy configuration
+        run: |
+          cp -f /opt/deployment-specific-files/chain.pem ./caddy/certs
+          cp -f /opt/deployment-specific-files/priv.pem ./caddy/certs
+          cp -f /opt/deployment-specific-files/.env ./
+
+      - name: Build
+        env:
+          # replace None with an empty string
+          BUILD_FLAG: ${{ github.event.inputs.build-flag != 'None' && github.event.inputs.build-flag || '' }}
+        run: docker-compose -f docker-compose.ssl.local.yml build $BUILD_FLAG
+
+      - name: Download DB dump
+        id: get-db
+        if: ${{ github.event.inputs.restore-db }}
+        uses: ./.github/actions/get-db
+        with:
+          workflow: ${{ github.event.inputs.environment == 'stage' && 'restore-sanitized-db.yml' || 'backup-db.yml' }}
+          zip-password: ${{ github.event.inputs.environment == 'stage' && secrets.STAGING_BACKUP_PASSWORD || secrets.BACKUP_PASSWORD }}
+
+      - name: Copy everything to /orbitar
+        run: |
+          mkdir -p /orbitar
+          rm -rf /orbitar/*
+          cp -r ./* /orbitar/
+          cd /orbitar
+
+      - name: Delete volumes
+        if : ${{ github.event.inputs.delete-volumes }}
+        run: docker-compose -f docker-compose.ssl.local.yml down -v
+
+      - name: Restore DB
+        if : ${{ github.event.inputs.restore-db }}
```

Comment:
1. Space after `if`
2. [When you use expressions in an if conditional, you may omit the expression syntax (${{ }})](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif)
3. Workflow doesn't seem to respect the boolean flags, always restores DB, always deletes volumes, always clears Redis.

Reply:
> Workflow doesn't seem to respect the boolean flags, always restores DB, always deletes volumes, always clears Redis.

Noticed that as well. Either the `gh workflow run` is broken and it will work once committed to `dev` and triggered via UI, or boolean params are not supported as an input type (weird).